csrc/mmdeploy/preprocess/transform/letter_resize.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) OpenMMLab. All rights reserved.
+
+#include <algorithm>
+#include <array>
+
+#include "mmdeploy/core/tensor.h"
+#include "mmdeploy/core/utils/formatter.h"
+#include "mmdeploy/operation/managed.h"
+#include "mmdeploy/operation/vision.h"
+#include "mmdeploy/preprocess/transform/tracer.h"
+#include "mmdeploy/preprocess/transform/transform.h"
+
+using namespace std;
+
+namespace mmdeploy::transform {
+
+class LetterResize : public Transform {
+ public:
+  explicit LetterResize(const Value& args) {
+    keep_ratio_ = args.value<bool>("keep_ratio", false);
+    if (args.contains("scale")) {
+      if (args["scale"].is_number_integer()) {
+        auto size = args["scale"].get<int>();
+        img_scale_ = {size, size};
+      } else if (args["scale"].is_array()) {
+        if (args["scale"].size() != 2) {
+          MMDEPLOY_ERROR("'scale' expects an array of size 2, but got {}", args["scale"].size());
+          throw std::length_error("'scale' expects an array of size 2");
```

Comment:
Let's use `throw_exception(eInvalidArgument);` instead of `throw std::length_error`